File: informatykaEuropejczyka/Rozdzial1/prog1_4.py
#algorytm "delty" rozwiazujacy rownanie kwadratowe

from math import *

def rownanie_kwadratowe (a, b, c):
    if a==0:
        print("to nie jest rownanie kwadratowe")
    else: 
        delta=b*b-4*a*c
        if delta<0:
            print("rownanie nie ma pierwiastkow")
        elif delta==0:
            x1=-b/(2*a)
            print("x = ",x1)
        else:    
            x1=(-b-sqrt(delta))/(2*a)
            x2=(-b+sqrt(delta))/(2*a)
            print("x1 = ",x1,"\tx2 = ",x2)

#rownanie_kwadratowe(-2,2,4)

File: informatykaEuropejczyka/Rozdzial1/prog1_4.cpp
//algorytm "delty" rozwiazujacy rownanie kwadratowe
#include <iostream>
#include <cmath>
using namespace std;
int main()
{
 double a, b, c, delta, x1, x2;
 cout<<"podaj wspolczynniki rownania a, b, c:"<<endl;
 cin>>a>>b>>c;
 if (a==0) cout<<"to nie jest rownanie kwadratowe"<<endl;
 else
 {
  delta=b*b-4*a*c;
  if (delta<0) cout<<"rownanie nie ma pierwiastkow"<<endl;
  else if (delta==0) {x1=-b/(2*a); cout<<"x="<<x1<<endl;}
       else
       {
        x1=(-b-sqrt(delta))/(2*a);
        x2=(-b+sqrt(delta))/(2*a);
        cout<<"x1="<<x1<<"\tx2="<<x2<<endl;
       }
 }
 return 0;
}
